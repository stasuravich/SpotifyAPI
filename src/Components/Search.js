import React, {useRef, memo} from 'react';
import '../Css/Search.css';

const Search = memo(function Search(props){
  //console.log(props.class);
  const inputRef = useRef();

  props.alerter(inputRef);

  return(
    <div className= {props.class} >
      <input ref = {inputRef} type="text" placeholder = {props.placeh} onChange={(e) => props.setQuery(e.target.value)}></input>
    </div>
  )
})

export default Search;
