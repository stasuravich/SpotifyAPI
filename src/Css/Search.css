.SearchPlaylist{
  padding-top: 10px;
  order: 1;
  padding-left: 250px;
  visibility: hidden;
}

.SearchOnline{
  padding-top: 10px;
  order: 3;
  padding-right: 250px;
}
