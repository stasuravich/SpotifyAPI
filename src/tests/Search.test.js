import React from 'react';
import Search from '../Components/Search';
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";

describe("rendering components", ()=> {
  it("renders Search without crashing", ()=>{
    shallow(<Search/>);
  });
})

describe("snapshots", () => {
  it("Search snapshots", () => {
    const tree= shallow(<Search/>);
    expect(toJson(tree)).toMatchSnapshot();
  });
})
