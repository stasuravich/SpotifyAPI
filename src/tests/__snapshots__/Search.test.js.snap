// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Search snapshots 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
  />
</div>
`;
